dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -511,6 +548,57 @@ private DhisDataValueStatusResponse importDataValues(Settings settings, String u
             String msg = String.format("Error parsing response from uri: %s, exception: %s", uri, e.toString());
             statusMessageService.warn(msg, MODULE_NAME);
             throw new DhisWebException(msg, e);
+        } finally {
+            closeResponse(response);
+        }
+    }
+
+    private void closeResponse(CloseableHttpResponse response) {
+        try {
+            if (response != null) {
+                EntityUtils.consume(response.getEntity());
+                response.close();
+            }
+        } catch (IOException e) {
+            LOGGER.error("Error trying to close response: " + e.getMessage());
+        }
+    }
+
+    @PreDestroy
+    public void cleanUp() throws IOException {
+        client.close();
+        poolingHttpClientConnectionManager.close();
+    }
+
+    private final class IdleConnectionMonitorThread extends Thread {
+        private final HttpClientConnectionManager connMgr;
+        private volatile boolean shutdown;
+
+        private IdleConnectionMonitorThread(PoolingHttpClientConnectionManager connMgr) {
```

Comment:
Fine, this can stay. This highlights the issues people have with http components though - complicated to use.